libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -36,21 +36,53 @@ using namespace solidity::frontend;
 namespace
 {
 
-void copyMissingTags(StructurallyDocumentedAnnotation& _target, set<CallableDeclaration const*> const& _baseFunctions)
+void copyMissingTags(set<CallableDeclaration const*> const& _baseFunctions, StructurallyDocumentedAnnotation& _target, CallableDeclaration const* _declaration = nullptr)
 {
 	if (_baseFunctions.size() != 1)
 		return;
 
 	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>((*_baseFunctions.begin())->annotation());
 
-	set<string> existingTags;
-
-	for (auto const& iterator: _target.docTags)
-		existingTags.insert(iterator.first);
-
-	for (auto const& [tag, content]: sourceDoc.docTags)
-		if (tag != "inheritdoc" && !existingTags.count(tag))
-			_target.docTags.emplace(tag, content);
+	for (auto it = sourceDoc.docTags.begin(); it != sourceDoc.docTags.end();)
+	{
+		string const& tag = it->first;
+		// Don't copy tag "inheritdoc" or already existing tags
+		if (tag == "inheritdoc" || _target.docTags.count(tag))
+		{
+			it++;
+			continue;
+		}
+
+		size_t n = 0;
+		// Iterate over all values of the current tag (it's a multimap)
+		for (auto next = sourceDoc.docTags.upper_bound(tag); it != next; it++, n++)
+		{
+			DocTag const& content = it->second;
+			DocTag newContent = content;
+			newContent.inherited = true;
+
+			// Update the parameter name for @return tags
+			if (_declaration && tag == "return")
+			{
+				size_t docParaNameEndPos = content.content.find_first_of(" \t");
+				string const docParameterName = content.content.substr(0, docParaNameEndPos);
+
+				if (docParameterName != _declaration->returnParameters().at(n)->name())
+				{
+					bool baseHasNoName = (*_baseFunctions.begin())->returnParameters().at(n)->name().empty();
+					string const paramName = _declaration->returnParameters().at(n)->name();
+					newContent.content =
+						(paramName.empty() ? "" : std::move(paramName) + " ") + (
+							string::npos == docParaNameEndPos || baseHasNoName ?
```

Comment:
What if we have `@returns x` (without description)?

Reply:
in that case it is assumed that `x` is the description and the position is used to identify which return parameter it is referring to.

Reply:
in terms of code, `returns` is not part of `content`, so the `docParaNameEndPos` would be `npos` as it couldn't find any whitespace or `\t`.

Reply:
Indeed, but then it would not substitute the name even though it should, right?

Reply:
It wouldn't and I don't think it should either as it isn't detected as "name" but as "description" instead.. either way it wouldn't have much use to have a name without a description OR a description that is just the name...

Reply:
Can you add a test for this case?